ios/RCCommonFunctionality.m

```diff
@@ -115,28 +140,35 @@ + (void)purchaseProduct:(NSString *)productIdentifier completionBlock:(RCHybridR
             }, nil);
         }
     };
-
+    
     [RCPurchases.sharedPurchases productsWithIdentifiers:@[productIdentifier]
                                          completionBlock:^(NSArray<SKProduct *> *_Nonnull products) {
-                                             NSMutableDictionary *productsByIdentifiers = [NSMutableDictionary new];
-                                             for (SKProduct *p in products) {
-                                                 productsByIdentifiers[p.productIdentifier] = p;
-                                             }
-                                             if (productsByIdentifiers[productIdentifier]) {
-                                                 [RCPurchases.sharedPurchases purchaseProduct:productsByIdentifiers[productIdentifier]
-                                                                          withCompletionBlock:completionBlock];
-                                             } else {
-                                                 NSError *error = [NSError errorWithDomain:RCPurchasesErrorDomain
-                                                                                      code:RCProductNotAvailableForPurchaseError
-                                                                                  userInfo:@{
-                                                                                          NSLocalizedDescriptionKey: @"Couldn't find product."
-                                                                                  }];
-                                                 completion(nil, [self payloadForError:error withExtraPayload:@{@"userCancelled": @(NO)}]);
-                                             }
-                                         }];
+        NSMutableDictionary *productsByIdentifiers = [NSMutableDictionary new];
```

Comment:
I may be wrong here, but it looks like the only use for this dictionary is to check if `productsByIdentifiers[productIdentifier]` exists and get it? if so, maybe just do a regular lookup instead

Reply:
Yes, I don't know why this is a dictionary